src/ralph/dashboards/models.py

```diff
@@ -283,3 +326,38 @@ def render(self, **context):
         if not renderer:
             raise RuntimeError('Wrong renderer.')
         return renderer(self).render(context)
+
+    def normalize_changelist_value(self, graph_item):
+        field = get_field_by_relation_path(
+            self.model.model_class(),
+            self.params['labels'].split(GRAPH_QUERY_SEP)[0]
+        )
+        if isinstance(field, ChoiceField):
+            choices = field.choice_class()
+            try:
+                graph_item = [
+                    i[0] for i in choices if i[1] == graph_item
+                ].pop()
+            except IndexError:
+                # NOTE(romcheg): Choice not found for the filter value.
+                #                Leaving it as is.
+                pass
+        elif isinstance(field, BooleanField):
+            graph_item = field.to_python(graph_item)
+        return graph_item
+
+    def get_queryset_for_filter(self, queryset, value):
+        filter_key = self.changelist_filter_key
+        value = self.normalize_changelist_value(value)
+        if self.custom_changelist_model:
+            value_param = self.params['target'].get('value', 'id')
+            values = self.build_queryset(annotated=False).filter(
```

Comment:
shouldn't queryset be passed here to `build_queryset`?

Reply:
only for line 359 is necessary